lib/syskit/gui/ide.rb

```diff
@@ -100,8 +100,8 @@ def reload_models
 
                 model_browser.registered_exceptions.clear
                 Roby.app.clear_exceptions
-                Roby.app.cleanup
                 Roby.app.clear_models
```

Comment:
Inverting is not really possible, the app is still "active" before cleanup, we can't be clearing the models.

Turns out that `cleanup` calls `clear_models`. We can call `cleanup_user_lifecycle_hooks` for the clear models hooks only after we have called clear_models (https://github.com/rock-core/tools-roby/blob/master/lib/roby/app.rb#L943)

Reply:
Ping @caioaamaral  ?

Reply:
Just to be sure, you mean:

 ```diff
 diff --git a/lib/syskit/gui/ide.rb b/lib/syskit/gui/ide.rb
index f45ce841..72a207f6 100644
--- a/lib/syskit/gui/ide.rb
+++ b/lib/syskit/gui/ide.rb
@@ -100,8 +100,9 @@ def reload_models
 
                 model_browser.registered_exceptions.clear
                 Roby.app.clear_exceptions
-                Roby.app.clear_models
                 Roby.app.cleanup
+                Roby.app.clear_models
+                Roby.app.cleanup_user_lifecycle_hooks
                 Roby.app.clear_config
                 Roby.app.setup
                 # HACK: reload_models calls Orocos.clear, which actually

diff --git a/lib/roby/app.rb b/lib/roby/app.rb
index f5ae1fa7..5e19f560 100644
--- a/lib/roby/app.rb
+++ b/lib/roby/app.rb
@@ -940,7 +940,7 @@ def cleanup
             cleanup_user_lifecycle_hooks(require_handlers)
             cleanup_user_lifecycle_hooks(controllers)
             cleanup_user_lifecycle_hooks(action_handlers)
-            cleanup_user_lifecycle_hooks(clear_models_handlers)
             cleanup_user_lifecycle_hooks(cleanup_handlers)
 
             call_plugins(:cleanup, self)

 ```

Reply:
Nope.

I mean, *in the cleanup handler*, you need to cleanup the lifecycle hooks after all the other handlers. And ideally not change anything on the syskit side.